.github/workflows/test-go.yml

```diff
@@ -502,8 +502,10 @@ jobs:
           fi
           echo "data-race-result=${result}" | tee -a "$GITHUB_OUTPUT"
       - name: Upload data race detector failure log
-        if: steps.data-race-check.outputs.data-race-result == 'failure'
-        uses: actions/upload-artifact@26f96dfa697d77e81fd5907df203aa23a56210a8 # v4.3.0
+        if: |
```

Comment:
This was the merge conflict. I took the behavior from main as it is more correct.